plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -488,49 +500,70 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
                         ("why",std::get<fc::exception_ptr>(response)->what()));
+                  fc_dlog(_trx_log, "[TRX_TRACE] Block ${block_num} for producer ${prod} is REJECTING tx: ${trx}",
+                          ("block_num", chain.head_block_num() + 1)("prod", get_pending_block_producer())
+                                ("trx", chain_plug->get_log_trx(trx->packed_trx()->get_transaction())));
+                  fc_dlog(_trx_trace_failure_log, "[TRX_TRACE] Block ${block_num} for producer ${prod} is REJECTING tx: ${entire_trace}",
+                          ("block_num", chain.head_block_num() + 1)("prod", get_pending_block_producer())
+                                ("entire_trace", get_trace(response)));
                } else {
-                  fc_dlog(_trx_successful_trace_log, "[TRX_TRACE] Speculative execution is REJECTING tx: ${txid}, auth: ${a} : ${why} ",
-                          ("txid", trx->id())
-                          ("a", trx->packed_trx()->get_transaction().first_authorizer())
-                          ("why",std::get<fc::exception_ptr>(response)->what()));
                   fc_dlog(_trx_failed_trace_log, "[TRX_TRACE] Speculative execution is REJECTING tx: ${txid}, auth: ${a} : ${why} ",
                           ("txid", trx->id())
                           ("a", trx->packed_trx()->get_transaction().first_authorizer())
                           ("why",std::get<fc::exception_ptr>(response)->what()));
+                  fc_dlog(_trx_log, "[TRX_TRACE] Speculative execution is REJECTING tx: ${trx} ",
+                          ("trx", chain_plug->get_log_trx(trx->packed_trx()->get_transaction())));
+                  fc_dlog(_trx_trace_failure_log, "[TRX_TRACE] Speculative execution is REJECTING tx: ${entire_trace} ",
+                          ("entire_trace", get_trace(response)));
                }
             } else {
                 if (!trx->read_only) {
                	   _transaction_ack_channel.publish(priority::low, std::pair<fc::exception_ptr, packed_transaction_ptr>(nullptr, trx->packed_trx()));
                 }
 
                if (_pending_block_mode == pending_block_mode::producing) {
-                  fc_dlog(_trx_successful_trace_log, "[TRX_TRACE] Block ${block_num} for producer ${prod} is ACCEPTING tx: ${txid}, auth: ${a}",
+                  fc_dlog(_trx_successful_trace_log, "[TRX_TRACE] Block ${block_num} for producer ${prod} is ACCEPTING tx: ${txid}, auth: ${a}, cpu: ${cpu}",
                           ("block_num", chain.head_block_num() + 1)
                           ("prod", get_pending_block_producer())
                           ("txid", trx->id())
-                          ("a", trx->packed_trx()->get_transaction().first_authorizer()));
+                          ("a", trx->packed_trx()->get_transaction().first_authorizer())
+                          (("cpu"), trx->billed_cpu_time_us));
```

Comment:
```suggestion
                          ("cpu", trx->billed_cpu_time_us));
```

Reply:
Concerned that future updates to the macro might not interpret this extra `()` correctly.